build.psm1

```diff
@@ -1104,11 +1108,11 @@ function Start-PSPackage {
     Write-Verbose "Packaging RID: '$Runtime'; Packaging Configuration: '$Configuration'" -Verbose
 
     # Make sure the most recent build satisfies the package requirement
-    if (-not $Script:Options -or                                ## Start-PSBuild hasn't been executed yet
-        -not $Script:Options.CrossGen -or                       ## Last build didn't specify -CrossGen
-        $Script:Options.Runtime -ne $Runtime -or                ## Last build wasn't for the required RID
-        $Script:Options.Configuration -ne $Configuration -or    ## Last build was with configuration other than 'Release'
-        $Script:Options.Framework -ne "netcoreapp1.0") ## Last build wasn't for CoreCLR
+    if (-not $Script:Options -or                                                    ## Start-PSBuild hasn't been executed yet
+        ((-not $Script:Options.CrossGen) -AND -not ($Runtime -match "-x86")) -or    ## Last build didn't specify -CrossGen OR CrossGen check skipped since it is not supported for x86
```

Comment:
I disagree. If the user specifies -CrossGen and the command completes successfully, they will expect that the assembly was actually crossgen'd. Silently ignoring the flag will lead to confusion when they realize that it wasn't actually crossgen'd.
The way it is currently written, specifying -CrossGen for x86 will trigger a crossgen error. Once CrossGen works for x86, users can specify it and it will succeed and the check should be reset back to what it was before this change.